services/order.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, MovieSession
+
+
+def create_order(
+    tickets: list[dict], username: str, date_created: datetime | str = None
+) -> None:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        order = Order.objects.create(user=user)
+
+        if date_created:
+            order.created_at = date_created
+            order.save()
+
+        movie_sessions = {}
+
+        for ticket in tickets:
+            # Attempting to reduce number of DB calls a little.
+            # For some reason I wasn't able to use `bulk_create()` here. :(
+            if (session_id := ticket["movie_session"]) not in movie_sessions:
+                movie_sessions[session_id] = MovieSession.objects.get(
+                    id=session_id
+                )
+
+            Ticket.objects.create(
+                movie_session=movie_sessions[session_id],
```

Comment:
could you make just `movie_session_id=session_id`?
without taking data from db: `MovieSession.objects.get(...`

Reply:
For some reason my code wasn't working when I tried to do it when I was implementing the original solution. Now it does. There was probably a typo that I wasn't able to find despite re-reading the code many times. :|

Reply:
magic 🪄